### PR TITLE
Co deputies

### PR DESCRIPTION
**Pull request by jackgoodby**

### CODE REVIEW

File: src/AppBundle/Controller/CoDeputyController.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        $this->denyAccessUnlessGranted([EntityDir\User::ROLE_DEPUTY]);
+
+        $data = $this->deserializeBodyContent($request, [
+            'email' => 'notEmpty',
+        ]);
+
+        $loggedInUser = $this->getUser();
+        $newUser = new EntityDir\User();
+
+        $newUser->setFirstname('');
```
Comment (elvisciotti) [inline src/AppBundle/Controller/CoDeputyController.php]: the logic from here could go in a method of the userService

File: src/AppBundle/Controller/CoDeputyController.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        $newUser->setLastname('');
```
Comment (slizzio) [inline src/AppBundle/Controller/CoDeputyController.php]: wont a new User object have an empty first and last name? Not sure above 2 statements are required

File: src/AppBundle/Controller/SelfRegisterController.php [hunk 3]
```diff
@@ -48,6 +49,42 @@ public function register(Request $request)
...
+            $coDeputyVerified = $this->container->get('user_registration_service')->validateCoDeputy($selfRegisterData);
+            $this->get('logger')->warning('CasRec codeputy validation success: ', ['extra' => ['page' => 'codep_validation', 'success' => true] + $selfRegisterData->toArray()]);
+        } catch (\Exception $e) {
+            $this->get('logger')->warning('CasRec codeputy validation failed:', ['extra' => ['page' => 'codep_validation', 'success' => false] + $selfRegisterData->toArray()]);
+throw $e;
+        }
+
+return print_r(doctrineDebug::export($coDeputyVerified, 3),true);
+
+        return ['verified' => $coDeputyVerified];
```
Comment (slizzio) [inline src/AppBundle/Controller/SelfRegisterController.php]: above 6 lines need a reformat. remove last return statement

File: src/AppBundle/Entity/User.php [hunk 4]
```diff
@@ -986,6 +995,44 @@ public function getAgreeTermsUseDate()
...
+     * @return bool
+     */
+    public function isCoDeputy()
+    {
+        $isCoDeputy = false;
+        if ($this->isLayDeputy()) {
+            $client = $this->getFirstClient();
+            if (!empty($client)) {
+                $isCoDeputy = count($client->getUsers()) > 1;
+            }
```
Comment (slizzio) [inline src/AppBundle/Entity/User.php]: Maybe throw an exception if client not set?

File: src/AppBundle/Controller/SelfRegisterController.php (hunk 3 above)
Comment (jackgoodby) [reply src/AppBundle/Controller/SelfRegisterController.php]: that was debug! good spot

**Pull request merged**